compaction/time_window_compaction_strategy.cc

```diff
@@ -433,15 +433,22 @@ time_window_compaction_strategy::get_buckets(std::vector<shared_sstable> files,
     return std::make_pair(std::move(buckets), max_timestamp);
 }
 
-static std::ostream& operator<<(std::ostream& os, const std::map<timestamp_type, std::vector<shared_sstable>>& buckets) {
-    os << "  buckets = {\n";
-    for (auto& bucket : buckets | boost::adaptors::reversed) {
-        os << format("    key={}, size={}\n", bucket.first, bucket.second.size());
-    }
-    os << "  }\n";
-    return os;
 }
 
+template <>
+struct fmt::formatter<std::map<sstables::timestamp_type, std::vector<sstables::shared_sstable>>> {
```

Comment:
Doesn't format provide formatters for containers?

Reply:
actually, we have two set of formatter formatting `std::map`. one from `utils/to_string.hh`, the other from `fmt/ranges.h`. but in this case, we customize the formatter by adding, for instance, `buckets = ` at the beginning of the output, and print the values like `key={key}, size={value.size)`. this is something neither of the homebrew generic formatter or the fmt/ranges.hh can offer.

hence this specialization.

Reply:
I see, fair enough.